### ISSUE TITLE
AttributeError in tracking extension

### ISSUE THREAD

**Issue by nisha-nec** [ckan/ckan#7964]
## CKAN version
Master

## Describe the bug
 I set up the CKAN master source code.  In ckan logs, I am receiving below AttributeError.

`2023-12-06 10:49:52,378 INFO  [ckan.config.middleware.flask_app] 500 Internal Server Error: The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.
2023-12-06 10:49:52,523 ERROR [ckan.config.middleware.flask_app] Request finalizing failed with an error while handling an error
Traceback (most recent call last):
  File "/usr/lib/ckan/default/lib/python3.10/site-packages/flask/app.py", line 2190, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/lib/ckan/default/lib/python3.10/site-packages/flask/app.py", line 1487, in full_dispatch_request
    return self.finalize_request(rv)
  File "/usr/lib/ckan/default/lib/python3.10/site-packages/flask/app.py", line 1508, in finalize_request
    response = self.process_response(response)
  File "/usr/lib/ckan/default/lib/python3.10/site-packages/flask/app.py", line 2002, in process_response
    response = self.ensure_sync(func)(response)
  File "/usr/lib/ckan/default/src/ckan/ckanext/tracking/middleware.py", line 35, in track_request
    key = h.update(key.encode()).hexdigest()
AttributeError: 'NoneType' object has no attribute 'hexdigest'`

Due to this error, the 'recent_view' count is not incremented after executing the tracking collect commands.

### Steps to reproduce
1. Setup the master source code.
2. login on the CKAN GUI.
3. create the dataset

### Expected behavior
The error should not be occur. 

### Investigation
As per my investigation, the root cause of this error is the update method updates the hash object and returns None. When it  call hexdigest() on the result of update, it's attempting to call hexdigest() on the return value, which is None. This results in the 'NoneType' object has no attribute 'hexdigest' error.

This error is occurred at the following path "ckan/ckanext/tracking/middleware.py", line 35, in track_request".

issue in below two lines:

```
h = hashlib.new('md5', usedforsecurity=False)  # type: ignore
key = h.update(key.encode()).hexdigest()
```
By replacing,  above these two lines with the below one, issue will be resolved. 

`key = hashlib.md5(key.encode()).hexdigest()`

**Comment by pdelboca** [comment]
Hello @nisha-nec ! Thanks for reporting this.

This will be fixed in #7967 :)

**Comment by nisha-nec** [comment]
I am closing this issue, as issue is resolved.
